cupy/sorting/sort.py

```diff
@@ -20,7 +28,54 @@ def sort(a):
     return ret
 
 
-# TODO(okuta): Implement lexsort
+def lexsort(keys):
+    """Perform an indirect sort using an array of keys.
+
+    Args:
+        keys (cupy.ndarray): ``(k, N)`` array containing ``k`` ``(N,)``-shaped
+            arrays. The ``k`` different "rows" to be sorted. The last row is
+            the primary sort key.
+
+    Returns:
+        cupy.ndarray: Array of indices that sort the keys.
+
+    .. note::
+        For its implementation reason, ``cupy.lexsort`` currently supports only
+        keys with their rank of one or two and does not support ``axis``
+        parameter that ``numpy.lexsort`` supports.
+
+    .. seealso:: :func:`numpy.lexsort`
+
+    """
+
+    # TODO(takagi): Support axis argument.
+
+    if keys.ndim == ():
+        msg = 'need sequence of keys with len > 0 in lexsort'
+        raise TypeError(msg)    # as numpy.lexsort() raises
+
+    if keys.ndim == 1:
+        return 0
+
+    # TODO(takagi): Support ranks of three or more.
+    if keys.ndim > 2:
+        msg = ('Keys with the rank of three or more is '
+               'not supported in lexsort')
+        raise ValueError(msg)
+
+    idx_array = cupy.ndarray(keys._shape[1:], dtype=numpy.int64)
+    k = keys._shape[0]
+    n = keys._shape[1]
+
+    try:
+        thrust.lexsort(keys.dtype, idx_array.data.ptr, keys.data.ptr, k, n)
+    except NameError:
+        msg = ('Thrust is needed to use cupy.lexsort. Please install CUDA '
+               'Toolkit with Thrust then reinstall CuPy after '
+               'uninstalling it.')
+        raise RuntimeError(msg)
```

Comment:
- Ditto (temporary variable is not needed)
- Rather than catching `NameError`, our convention is to make a flag to indicate if thrust is available. Please see how cusolver is imported in `cupy/cuda/__init__.py`.